src/back/glsl.rs

```diff
@@ -1596,6 +1620,13 @@ fn builtin_to_glsl(builtin: BuiltIn) -> &'static str {
     }
 }
 
+fn get_base_ty<'a>(types: &'a Arena<Type>, ty: &'a TypeInner) -> &'a TypeInner {
+    match ty {
+        TypeInner::Pointer { base, .. } => get_base_ty(types, &types[*base].inner),
```

Comment:
this function specifically works with pointers? why did you need it?

Reply:
I had some issues where i got pointers where other types where expected this is a temporary solution

Reply:
Please tell more. I thought we fixed this in #100. If not, this should be fixed in the front-ends, not worked around here.

Reply:
I just checked right now `test-data/spv/cube.frag.spv` and the frontend is producing multiple loads to globals